utils/ccdatofhir/mapping_resource_allergy_intolerance.bal

```diff
@@ -0,0 +1,189 @@
+        if playingEntityCodeableConcept is r4:CodeableConcept {
+            allergyIntolerance.code = playingEntityCodeableConcept;
+        }
+
+        xml observationElement = entryRelationshipElement/<v3:observation|observation>;
+        xml observationValueElement = observationElement/<v3:value|value>;
+        xml observationIdElement = observationElement/<v3:id|id>;
+
+        r4:CodeableConcept? clinicalStatus = mapCcdaCodingtoFhirCodeableConcept(observationValueElement);
+        if clinicalStatus is r4:CodeableConcept {
+            allergyIntolerance.clinicalStatus = clinicalStatus;
+        }
+
+        r4:CodeableConcept? manifestation = mapCcdaCodingtoFhirCodeableConcept(observationValueElement);
+        if manifestation is r4:CodeableConcept {
+            r4:AllergyIntoleranceReaction reaction = {
+                manifestation: [manifestation]
+            };
+
+            string|error? typeCode = entryRelationshipElement.typeCode;
+            if typeCode is string && typeCode == "MFST" {
+                string|error? reactionId = observationIdElement.root;
+                if reactionId is string {
+                    reaction.id = reactionId;
+                } else {
+                    log:printDebug("reactionId is not available", reactionId);
+                }
+            } else if typeCode is string && typeCode != "MFST" {
+                log:printDebug(string `typeCode ${typeCode} is not MFST`);
+            } else if typeCode is error {
+                log:printDebug("typeCode is not available", typeCode);
+            }
+            allergyIntolerance.reaction = [reaction];
+        }
+        return allergyIntolerance;
+    } else {
+        log:printDebug("AllergyIntolerance not available");
+        return ();
+    }
+}
+
+isolated function mapCcdaValueToFhirAllergyIntoleranceCategory(xml valueElement) returns r4:AllergyIntoleranceCategory? {
+    string|error? codeVal = valueElement.code;
+    if codeVal is string {
+        match codeVal {
+            "235719002" => {
+                return r4:CODE_CATEGORY_FOOD;
+            }
+            "416098002" => {
+                return r4:CODE_CATEGORY_MEDICATION;
+            }
+            "414285001" => {
+                return r4:CODE_CATEGORY_FOOD;
+            }
+            "418471000" => {
+                return r4:CODE_CATEGORY_FOOD;
+            }
+            "419511003" => {
+                return r4:CODE_CATEGORY_MEDICATION;
+            }
+            "59037007" => {
+                return r4:CODE_CATEGORY_MEDICATION;
+            }
+            _ => {
+                log:printDebug("matching code value not available");
+                return ();
+            }
+        }
+    } else {
+        log:printDebug("code value not available", codeVal);
+        return ();
+    }
+}
+
+isolated function mapCcdaValueToFhirAllergyIntoleranceType(xml valueElement) returns r4:AllergyIntoleranceType? {
```

Comment:
Can we combine this function with `mapCcdaValueToFhirAllergyIntoleranceCategory` and return a `[r4:AllergyIntoleranceCategory, AllergyIntoleranceType]?` ?

Reply:
In future extensions of this lib, there can be instances where these 2 result values are used individually in several places according to the mapping spec we follow for this implementation. In such cases, a combined method returning 2 mapped results (but only one is being used) unnecessarily would cause Ballerin unused variable error.